### PR TITLE
dependency auditing implementation

### PR DESCRIPTION
**Pull request by rav-pradhan**
### What

Implement dependency auditing with Nancy

### How to review

- Check changes align with other implementations for dependency auditing within our services
- Ensure correct permissions are set for the `audit.sh` file (chmod 755)

### Who can review

Anyone

### CODE REVIEW

**Review by DavidSubiros — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -11,15 +11,31 @@ LDFLAGS=-ldflags "-w -s -X 'main.Version=${VERSION}' -X 'main.BuildTime=$(BUILD_
 
 DATABASE_ADDRESS?=bolt://localhost:7687
 
+PHONY: all
```
Comment (CarlHembrough) [inline Makefile:14]: Does `PHONY` need to be prefixed with a `.`? Most repos have the prefix, though I also noticed that the `dp-api-router` doesn't.

**Review by DavidSubiros — CHANGES_REQUESTED** [review]
sorry, didn't see @CarlHembrough  comment before

**Review by DavidSubiros — CHANGES_REQUESTED** [review]
Same comment as noted by @CarlHembrough in https://github.com/ONSdigital/dp-filter-api/pull/158

```Does PHONY need to be prefixed with a .? Most repos have the prefix, though I also noticed that the dp-api-router doesn't.```

File: Makefile (hunk 1 above)
Comment (rav-pradhan) [reply Makefile:14]: I think it does need to be prefixed with a `.` - I will update here and in `dp-api-router`, too

**Review by DavidSubiros — APPROVED** [review]

**Pull request merged**